maat-scheduled-tasks/src/main/java/uk/gov/justice/laa/maat/scheduled/tasks/service/XhibitDataService.java

```diff
@@ -0,0 +1,158 @@
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.CopyObjectRequest;
+import software.amazon.awssdk.services.s3.model.CopyObjectResponse;
+import software.amazon.awssdk.services.s3.model.DeleteObjectRequest;
+import software.amazon.awssdk.services.s3.model.DeleteObjectResponse;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.InvalidObjectStateException;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request;
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response;
+import software.amazon.awssdk.services.s3.model.NoSuchKeyException;
+import software.amazon.awssdk.services.s3.model.ObjectNotInActiveTierErrorException;
+import uk.gov.justice.laa.maat.scheduled.tasks.config.XhibitConfiguration;
+import uk.gov.justice.laa.maat.scheduled.tasks.dto.XhibitRecordSheetDTO;
+import uk.gov.justice.laa.maat.scheduled.tasks.enums.RecordSheetStatus;
+import uk.gov.justice.laa.maat.scheduled.tasks.enums.RecordSheetType;
+import uk.gov.justice.laa.maat.scheduled.tasks.exception.XhibitDataServiceException;
+import uk.gov.justice.laa.maat.scheduled.tasks.responses.GetRecordSheetsResponse;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class XhibitDataService {
+
+    private final S3Client s3Client;
+
+    private final XhibitConfiguration xhibitConfiguration;
+
+    public GetRecordSheetsResponse getRecordSheets(RecordSheetType recordSheetType, String continuationToken) {
+        try {
+            GetRecordSheetsResponse recordSheetsResponse = new GetRecordSheetsResponse();
+
+            String objectKeyPrefix = getPrefixString(recordSheetType);
+
+            ListObjectsV2Request.Builder listObjectsRequestBuilder = ListObjectsV2Request.builder()
+                .bucket(xhibitConfiguration.getS3DataBucketName())
+                .prefix(objectKeyPrefix);
+
+            if (continuationToken != null) {
+                listObjectsRequestBuilder.continuationToken(continuationToken);
+            }
+
+            ListObjectsV2Response listObjectsResponse = s3Client.listObjectsV2(
+                listObjectsRequestBuilder.build());
+
+            if (listObjectsResponse.contents().isEmpty()) {
+                return recordSheetsResponse;
+            }
+
+            listObjectsResponse.contents().forEach(s3Object -> {
+                try {
+                    ResponseBytes<GetObjectResponse> objectResponse = s3Client.getObjectAsBytes(
+                        GetObjectRequest.builder()
+                            .bucket(xhibitConfiguration.getS3DataBucketName())
+                            .key(s3Object.key())
+                            .build());
+
+                    String trialDataXml = objectResponse.asUtf8String();
+
+                    recordSheetsResponse.getRetrievedRecordSheets()
```

Comment:
Instead of storing all the records in a single list, should we be writing individual records to DB to avoid OutOfMemoryErrors?